DDG4/lcio/LCIOEventReader.cpp

```diff
@@ -146,11 +146,10 @@ LCIOEventReader::readParticles(int event_number,
       vtx->y = p->vsy ;
       vtx->z = p->vsz ;
       vtx->time = p->time ;
-      haveVertex = false ;
+      haveVertex = true ;
```

Comment:
This does not change anything. In line 86 `bool haveVertex = true ;` subsequently if the `if( !haveVertex && genStatus == 1 ){` is always false as describes in [CID 1402993](https://scan7.coverity.com/reports.htm#v17832/p11968/fileInstanceId=9708936&defectInstanceId=2393793&mergedDefectId=1402993). You have to change `if( !haveVertex && genStatus == 1 ){` to `if( haveVertex && genStatus == 1 ){` to address this.